apps/condo/domains/miniapp/schema/SendB2CAppPushMessageService.js

```diff
@@ -54,59 +72,80 @@ const ERRORS = {
+        messageForUser: `api.miniapp.${SERVICE_NAME}.WRONG_SENDER_FORMAT`,
     },
     APP_IN_BLACK_LIST: {
         mutation: SERVICE_NAME,
         code: FORBIDDEN,
         type: APP_BLACK_LIST_ERROR,
-        message: 'Notification not send because app of message added in MessageAppBlackList',
+        message: 'Could not send notifications for the blacklisted app',
+        messageForUser: `api.miniapp.${SERVICE_NAME}.APP_IN_BLACK_LIST`,
     },
 }
 
-const SendAppPushMessageService = new GQLCustomSchema('SendAppPushMessageService', {
+const SendB2CAppPushMessageService = new GQLCustomSchema('SendB2CAppPushMessageService', {
     types: [
         {
             access: true,
-            type: `enum SendAppPushMessageType { ${VOIP_INCOMING_CALL_MESSAGE_TYPE} ${B2C_APP_MESSAGE_PUSH_TYPE} }`,
+            type: `enum SendB2CAppPushMessageType { ${VOIP_INCOMING_CALL_MESSAGE_TYPE} ${B2C_APP_MESSAGE_PUSH_TYPE} }`,
         },
         {
             access: true,
-            type: 'input SendAppPushMessageData { body: String!, title: String, B2CAppContext: String }',
+            type: 'input SendB2CAppPushMessageData { body: String!, title: String, B2CAppContext: String }',
         },
         {
             access: true,
-            type: 'input SendAppPushMessageInput { dv: Int!, sender: SenderFieldInput!, app: B2CAppWhereUniqueInput!, user: UserWhereUniqueInput!, type: SendAppPushMessageType!, data: SendAppPushMessageData! }',
+            type: 'input SendB2CAppPushMessageInput { ' +
+                    'dv: Int!, ' +
+                    'sender: SenderFieldInput!, ' +
+                    'app: B2CAppWhereUniqueInput!, ' +
+                    'user: UserWhereUniqueInput!, ' +
+                    'resident: ResidentWhereUniqueInput!, ' +
+                    'type: SendB2CAppPushMessageType!, ' +
+                    'data: SendB2CAppPushMessageData! ' +
+                '}',
         },
         {
             access: true,
-            type: 'type SendAppPushMessageOutput { id: String!, status: String! }',
+            type: 'type SendB2CAppPushMessageOutput { id: String!, status: String! }',
         },
     ],
     
     mutations: [
         {
-            access: access.canSendAppPushMessage,
-            schema: 'sendAppPushMessage(data: SendAppPushMessageInput!): SendAppPushMessageOutput',
-            resolver: async (parent, args, context, info, extra = {}) => {
+            access: access.canSendB2CAppPushMessage,
+            schema: 'sendB2CAppPushMessage(data: SendB2CAppPushMessageInput!): SendB2CAppPushMessageOutput',
+            resolver: async (parent, args, context) => {
                 const { data: argsData } = args
-                const { dv, sender, app, user, type, data, uniqKey } = argsData
+                const { dv, sender, app, user, resident, type, data, uniqKey } = argsData
                 const { B2CAppContext } = data
+                const residentId = get(resident, 'id')
```

Comment:
Whether resident object may not contain the given field? Why not just throw an error without additional db query right after this line?

Reply:
that left of previous state, when resident was non required field. fixed.